diff -urN xfce4-sensors-plugin-1.0.0/include/sensors-interface-common.h xfce4-sensors-plugin-1.0.0.new//include/sensors-interface-common.h
--- xfce4-sensors-plugin-1.0.0/include/sensors-interface-common.h	2010-03-28 18:11:33.000000000 +0000
+++ xfce4-sensors-plugin-1.0.0.new//include/sensors-interface-common.h	2011-02-17 11:59:15.459999142 +0000
@@ -31,7 +31,7 @@
 /* #include <glib/gprintf.h>  */
 
 /* Xfce includes */
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 /* Package includes */
 #include <types.h>
diff -urN xfce4-sensors-plugin-1.0.0/panel-plugin/sensors-plugin.c xfce4-sensors-plugin-1.0.0.new//panel-plugin/sensors-plugin.c
--- xfce4-sensors-plugin-1.0.0/panel-plugin/sensors-plugin.c	2010-03-28 18:14:19.000000000 +0000
+++ xfce4-sensors-plugin-1.0.0.new//panel-plugin/sensors-plugin.c	2011-02-17 11:57:46.630000004 +0000
@@ -41,7 +41,7 @@
 /* Xfce includes */
 #include <libxfce4util/libxfce4util.h>
 #include <libxfcegui4/libxfcegui4.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 /* Package includes */
 #include <configuration.h>
diff -urN xfce4-sensors-plugin-1.0.0/src/Makefile.am xfce4-sensors-plugin-1.0.0.new//src/Makefile.am
--- xfce4-sensors-plugin-1.0.0/src/Makefile.am	2009-11-06 11:02:11.000000000 +0000
+++ xfce4-sensors-plugin-1.0.0.new//src/Makefile.am	2011-02-17 12:13:21.563333271 +0000
@@ -27,7 +27,8 @@
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\" \
 	@LIBSENSORS_CFLAGS@	\
 	@LIBXFCE4UTIL_CFLAGS@ \
-	@LIBXFCEGUI4_CFLAGS@
+	@LIBXFCEGUI4_CFLAGS@ \
+	@LIBXFCE4PANEL_CFLAGS@
 
 # @GTK_CFLAGS@ 	\
 # @GLIB_CFLAGS@ \
