--- xfce4-sensors-plugin-1.0.0/lib/sensors-interface.c~	2011-04-21 11:00:03.512857739 +0200
+++ xfce4-sensors-plugin-1.0.0/lib/sensors-interface.c	2011-04-21 11:02:55.986062656 +0200
@@ -105,7 +105,7 @@
                 if (!notify_is_initted())
                     notify_init(PACKAGE); /* NOTIFY_APPNAME */
 
-                nn = notify_notification_new(summary, body, icon, NULL);
+                nn = notify_notification_new(summary, body, icon);
                 notify_notification_show(nn, &error);
                 #else
                 DBG("%s\n%s", summary, body);
