--- xfce4-sensors-plugin-1.2.0/lib/sensors-interface.c~	2011-05-09 18:22:12.000000000 +0200
+++ xfce4-sensors-plugin-1.2.0/lib/sensors-interface.c	2011-07-06 11:49:13.878027422 +0200
@@ -34,7 +34,7 @@
 #include <gtk/gtk.h>
 
 /* Xfce includes */
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 /* Package includes */
 #include <sensors-interface-common.h>
--- xfce4-sensors-plugin-1.2.0/panel-plugin/sensors-plugin.c~	2011-05-05 16:32:20.000000000 +0200
+++ xfce4-sensors-plugin-1.2.0/panel-plugin/sensors-plugin.c	2011-07-06 11:50:56.166527400 +0200
@@ -42,7 +42,7 @@
 #include <libxfce4util/libxfce4util.h>
 //#include <libxfcegui4/libxfcegui4.h>
 #include <libxfce4ui/libxfce4ui.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 /* Package includes */
 #include <configuration.h>
--- xfce4-sensors-plugin-1.2.0/src/callbacks.c~	2011-03-25 22:08:47.000000000 +0100
+++ xfce4-sensors-plugin-1.2.0/src/callbacks.c	2011-07-06 11:51:54.204527418 +0200
@@ -27,7 +27,7 @@
 #include <stdlib.h>
 
 /* Xfce includes */
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 /* Package includes */
 #include <sensors-interface.h>
--- xfce4-sensors-plugin-1.2.0/src/actions.c~	2011-03-25 22:08:47.000000000 +0100
+++ xfce4-sensors-plugin-1.2.0/src/actions.c	2011-07-06 11:52:56.042527552 +0200
@@ -28,7 +28,7 @@
 #include <gtk/gtk.h>
 
 /* Xfce includes */
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 /* Package includes */
 #include <middlelayer.h>
--- xfce4-sensors-plugin-1.2.0/src/interface.c~	2011-03-25 22:08:47.000000000 +0100
+++ xfce4-sensors-plugin-1.2.0/src/interface.c	2011-07-06 11:53:30.487027419 +0200
@@ -24,7 +24,7 @@
 #endif
 
 /* Xfce includes */
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 /* Package includes */
 #include <cpu.h>
